application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -77,16 +77,17 @@ private Cache<Long, Instant> createCooldownCache() {
     @Override
     public void onMessageContext(MessageContextInteractionEvent event) {
         long userID = event.getUser().getIdLong();
-        String reportedMessageTimestamp =
-                event.getTarget().getTimeCreated().format(DateTimeFormatter.ISO_DATE_TIME);
+        String reportedMessageTimestamp = event.getTarget().getTimeCreated().toInstant().toString();
 
-        if (handleIsOnCooldown(userID, event)) {
+        if (handleIsOnCooldown(event)) {
             return;
         }
         authorToLastReportInvocation.put(userID, Instant.now());
 
         String reportedMessage = event.getTarget().getContentRaw();
-        String reportedMessageUrl = event.getTarget().getJumpUrl();
+        String authorName = event.getTarget().getAuthor().getName();
+        String authorAvatarURL = event.getTarget().getAuthor().getAvatarUrl();
+        String authorID = event.getTarget().getAuthor().getId();
```

Comment:
which author? the user who reported (author of the "reason") or the author of the reported message.

prevent confusion and name it `reportedAuthorName` etc.

Reply:
(not done yet)